torch/testing/_internal/common_distributed.py

```diff
@@ -877,6 +888,81 @@ def _check_return_codes(self, elapsed_time) -> None:
     def is_master(self) -> bool:
         return self.rank == 0
 
+# Utility base class for distributed Multi Process Test cases
+# This abstracts the PG creation and deletion, the backends are selected based
+# on device type. The tests functions can be instantiated per device type using
+# common_device_type.instantiate_device_type_tests
+# other backends can add entry in backend() function
+class DistributedTestBase(MultiProcessTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self._spawn_processes()
+
+    def tearDown(self):
+        try:
+            os.remove(self.file_name)
+        except OSError:
+            pass
+
+    def backend(self, device) -> str:
+        if "cuda" in device:
+            return "nccl"
+        elif "hpu" in device :   # intel gaudi
+            return "hccl"
+        else :
+            return "gloo"
+
+    def device_handle(self, device) :
+        """
+        Get the torch module for the given device type.
+
+        This method is used to handle different device types in distributed testing.
+        It extracts the device name from the input string and returns the corresponding
+        torch module if available.
+
+        Args:
+            device (str): A string representing the device, e.g., "cuda:0", "cpu", "hpu".
+
+        Returns:
+            Union[torch.device, None]: The corresponding torch module if available,
+                                       or None if the device is not found but is an attribute of torch.
+
+        Raises:
+            RuntimeError: If the device library is not found in torch.
+
+        Example:
+            >>> self.device_handle("cuda:0")
+            <module 'torch.cuda' from '...'>
+        """
+        device_name = device.split(":")[0]
+        if hasattr(torch, device_name):
+            return getattr(torch, device_name, None)
+        else:
+            raise RuntimeError(
+                f"Device {device} library not found"
+            )
+
+    def create_pg(self, device):
+        num_visible_devices = self.device_handle(device).device_count()
+        store = torch.distributed.FileStore(self.file_name, num_visible_devices)
+        torch.distributed.init_process_group(
+            backend=self.backend(device),
+            world_size=self.world_size,
+            rank=self.rank,
+            store=store
+        )
+        if "nccl" in self.backend(device):
+            torch.cuda.set_device(self.rank)
+        return torch.distributed.distributed_c10d._get_default_group()
+
+    def rank_to_device(self, device):
+        num_visible_devices = self.device_handle(device).device_count()
+        return {i: [i % num_visible_devices] for i in range(self.world_size)}
+
+    def destroy_pg(self) -> None:
+        torch.distributed.barrier()
```

Comment:
Considering this is a base test class, can you please remove the `barrier()` call? We want to make sure our tests works as expected even without a barrier at the end. If a specific test needs a barrier, it should be called explicitly in that test.

Reply:
And if there is only one line left, maybe no more need to wrap it in a method?

Reply:
Done in commit: e97d3a38089a266cf7f37d69309e208d9c58ee0b